edx_django_utils/db/queryset_utils.py

```diff
@@ -0,0 +1,45 @@
+"""
+Utils related to QuerySets.
+"""
+
+
+def chunked_queryset(queryset, chunk_size=100):
+    """
+    Slice a queryset into chunks.
+
+    This utility method is helpful for large QuerySets. As it is not recommended to load
+    whole QuerySet data at a same time. Using this method we can iterate over the large QuerySets
+    and load only a chunked piece of the QuerySets at a time.
+
+    Source: https://www.djangosnippets.org/snippets/10599/
```

Comment:
I'd like @ormsbee review this and/or to point to docs he may have written on proper paging.

Here are the docs from this source:
> The function slices a queryset into smaller querysets containing chunk_size objects and then yield them. It is used to avoid memory error when processing huge queryset, and also database error due to that the database pulls whole table at once. Concurrent database modification wouldn't make some entries repeated or skipped in this process.

This scares me that this is a workaround for getting an anti-pattern to not break things, rather than a best practice that we want people to use. Note that we shouldn't be adding this to the library if we don't think this is a best practice.

Again, I think @ormsbee is in a better position to help determine if this should be a part of a best practice, or if not, how to suggest better alternatives.

Reply:
Where are we going to be using this to start?

Reply:
@ormsbee: I'm not sure about new uses, but you can see existing uses here: https://github.com/search?q=org%3Aedx+chunked_queryset&type=code

Reply:
@robrap: So it looks like this is being used in management commands, which makes sense.

I'm not super-thrilled by this code snippet in that it does its own `order_by`–it makes perfect sense given the implementation, but it's surprising behavior from the outside. I'd honestly just rather avoid using it if possible most of the time.

That being said, when memory usage actually is a problem (e.g. iterating through millions of records), this is about the only hammer to reach for. The normal solution for this is supposed to be [iterator()](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#iterator), but that only really does the cursor thing with Postgres and Oracle. The MySQL driver loads it all into memory anyway.

So I'm fine with its inclusion in the library, provided that there is proper guidance on when to use it and the caveats (e.g. "it throws away your sorting"). I wouldn't want this casually used, but in the circumstances that it's needed, it's probably better implemented once than to have three or four reimplemented versions of this floating around different repos in the codebase.

Reply:
I think we can be more aggressive with the default chunk size through. The default Django uses for its own `iterator()` method is 2000, so that's probably a decent starting point.

Reply:
Thanks @ormsbee. 
@HammadAhmadWaqas - please update both the default and the docs accordingly based on the notes above. Thank you!

Reply:
I have updated the PR.

@robrap @ormsbee 
Thank you so much for both of you for this detailed and quick review. This will help a lot to ship the dependent feature on time.